src/adapters/parsers.ts

```diff
@@ -0,0 +1,10 @@
+import { z } from "zod";
+
+import { StrictSchema } from "src/utils/type-safety";
+import { Env } from "src/domain";
+
+const envParser = StrictSchema<Env>()(z.object({ REACT_APP_INFURA_API_KEY: z.string() }));
+
+const ethereumAccountsParser = StrictSchema<string[]>()(z.array(z.string()));
```

Comment:
I don't agree with that approach. Because, if you need to use the parser more than one time you'll always need to write the `z.array()`... Also if the list of entities is not an array the parser could be different... Another thing that I don't like is to import `zod` outside of this module.

On what I agree is that we should have one parser for an entity and another one for the entity list (if needed), like this:

```ts
const ethereumAccountParser = StrictSchema<string>()(z.string());
const ethereumAccountsParser = StrictSchema<string[]>()(z.array(z.string()));
```

But yes, in this case as we are not using the single entity and it's very straightforward, I agree that it's not necessary.